oidc/viewsets.py

```diff
@@ -176,5 +197,43 @@ class UserModelOpenIDConnectViewset(BaseOpenIDConnectViewset):
 
     user_model = get_user_model()
 
-    def create_login_user(self, kwargs: dict):
-        return self.user_model.objects.create(**kwargs)
+    def create_login_user(self, user_data: dict):
+        return self.user_model.objects.create(**user_data)
+
+
+class RapidProOpenIDConnectViewset(BaseOpenIDConnectViewset):
+    """
+    OpenID Connect Viewset tailored to work with
+    RapidPro(https://github.com/rapidpro/rapidpro)
+    """
+
+    user_model = get_user_model()
+
+    def create_login_user(self, user_data: dict):
+        Org = importlib.import_module("temba").orgs.models.Org
+        timezone = importlib.import_module("pytz").timezone
+        org_name = user_data.pop("username")
+        user_data["username"] = user_data.get("email")
+
+        org_data = {
+            "name": org_name,
+            "slug": Org.get_unique_slug(org_name),
+            "brand": settings.DEFAULT_BRAND,
+            "timezone": timezone("UTC"),
+        }
+        user = self.user_model.objects.create(**user_data)
+
+        language = self.request.branding.get("language", settings.DEFAULT_LANGUAGE)
+        user_settings = user.get_settings()
+        user_settings.language = language
+        user_settings.save()
+
+        org_data.update({"created_by": user, "modified_by": user})
+        org = Org.objects.create(**org_data)
+        org.administrators.add(user)
+        branding = org.get_branding()
+        org.initialize(
+            branding=branding, topup_size=branding.get("welcome_topup", 1000)
+        )
```

Comment:
I've omitted the use of invitations here since I'm not entirely sure how that would work with OpenID Connect. Would the users have an invitation secret sent alongside the `id_token` ?

_[Invitation code](https://github.com/rapidpro/rapidpro/blob/ab1596268d7b257ad10ad198a44dbb1580549dd6/temba/orgs/views.py#L1757-L1769)_

Reply:
Have you tested? I would say let's test. Please create a new issue to address this in a new PR.